test/e2e/common/expansion.go

```diff
@@ -306,6 +308,119 @@ var _ = framework.KubeDescribe("Variable Expansion", func() {
 		// Pod should fail
 		testPodFailSubpath(f, pod)
 	})
+
+	/*
+		    Testname: var-expansion-subpath-lifecycle
```

Comment:
Can you add one more test case that:
1. Starts the container successfully with a expanded subpath
2. Change the value of the env variable
3. Restart the container
4. Validate that the subpath is still using the old value

Reply:
Added